layers-samples/eqrt-video.html

```diff
@@ -0,0 +1,268 @@
+    );
+    menuSystem.createButton('../media/textures/forward-button.png', () => { eqrtVideoElement.currentTime += 15; });
+    menuSystem.createButton('../media/textures/x-button.png', () => { xrSession.end() });
+    scene.addNode(menuSystem.getMenuBarNode());
+
+    function initXR() {
+      xrButton = new WebXRButton({
+        onRequestSession: onRequestSession,
+        onEndSession: onEndSession
+      });
+      document.querySelector('header').appendChild(xrButton.domElement);
+
+      if (navigator.xr) {
+        navigator.xr.isSessionSupported('immersive-vr').then((supported) => {
+          xrButton.enabled = supported;
+        });
+      }
+    }
+
+    function onRequestSession() {
+      if (!xrSession) {
+        navigator.xr.requestSession('immersive-vr', {
+          requiredFeatures: ['layers'],
+        }).then(onSessionStarted);
+      } else {
+        onEndSession();
+      }
+    }
+
+    function initGL() {
+      if (gl) { return; }
+      gl = createWebGLContext({ xrCompatible: true, webgl2: true, });
+      document.body.appendChild(gl.canvas);
+      gl.clearColor(0.0, 0, 0, 0.0);
+
+      function onResize() {
+        gl.canvas.width = gl.canvas.clientWidth * window.devicePixelRatio;
+        gl.canvas.height = gl.canvas.clientHeight * window.devicePixelRatio;
+      }
+      window.addEventListener('resize', onResize);
+      onResize();
+
+      renderer = new Renderer(gl);
+      scene.setRenderer(renderer);
+
+      // Util for rendering stereo layers
+      stereoUtil = new VRStereoUtil(gl);
+    }
+
+    function onSessionStarted(session) {
+      xrSession = session;
+      scene.inputRenderer.useProfileControllerMeshes(session);
+      session.addEventListener('end', onSessionEnded);
+
+      initGL();
+
+      xrFramebuffer = gl.createFramebuffer();
+      xrGLFactory = new XRWebGLBinding(session, gl);
+
+      let selected = document.getElementById("eqrtVideoSelect");
+      eqrtVideoElement = document.createElement('video');
+      eqrtVideoElement.loop = true;
+      eqrtVideoElement.crossOrigin = "anonymous";
+      eqrtVideoElement.preload = 'auto';
+      eqrtVideoElement.autoload = true;
+      eqrtVideoElement.src = selected.options[selected.selectedIndex].value;
+      eqrtVideoWidth = selected.options[selected.selectedIndex].getAttribute("data-video-width");
+      eqrtVideoHeight = selected.options[selected.selectedIndex].getAttribute("data-video-height");
+      let eqrtVideoAngle = selected.options[selected.selectedIndex].getAttribute("data-video-angle");
+      let eqrtVideoLayout = selected.options[selected.selectedIndex].getAttribute("data-video-layout");
+      let eqrtRadius = 10;
+      let eqrtVideoFPS = document.getElementById("eqrtVideoFPS").value;
+
+
+      eqrtVideoElement.play();
+      setInterval(function () {
+        if (eqrtVideoElement.readyState >= eqrtVideoElement.HAVE_CURRENT_DATA) {
+          eqrtVideoNeedsUpdate = true;
+        }
+      }, 1000 / eqrtVideoFPS);
```

Comment:
There's a bug here where if you start and exit a session, change the video framerate, then enter a new session the interval from the old one is still firing and you won't see any difference in the framerate. You could avoid this by cancelling the interval on end session.

Beyond that, though, I'm not entirely sure why this is needed? We could likely set this based on the video file being used and it would be fine.

Reply:
Good point, I changed it to read frame rate from video selection, and removed the frame rate input